orc8r/cloud/helm/orc8r/charts/certs/templates/certifier-certificate.yaml

```diff
@@ -0,0 +1,27 @@
+# Copyright 2021 The Magma Authors.
+
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+{{- if .Values.create }}
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: {{ .Release.Name }}-certifier-certificate
+spec:
+  secretName: "{{ .Release.Name }}-certifier-tls"
+  commonName: "certifier.{{ required "dnsDomain must be provided" .Values.dnsDomain }}"
+  subject:
+    countries:
+      - US
+  duration: 87600h
```

Comment:
These hard-coded 10-year values should probably be variables. Related: what is the max duration supported for this value?

This is probably also a good point to consider what happens after they expire. The problem is we provision gateways with the specific root and fluentd crts, so when they expire under our current architecture, shit goes down. Specifically

- Root
    - All gateways drop connection to Orc8r, requiring reprovisioning of all gateways in network
        - Solution 1: support in-band mechanism for Orc8r sending new rootCA.pem to all gateways (messy/what if a gateway was disconnected from Orc8r for long period of time)
        - Solution 2: remove support for self-signed certs -- this way, gateways wouldn't need to have a rootCA.pem, instead relying on OS-level support
- Fluentd
    - Same story as root but just for log aggregation. Probably same solution too
- Certifier
    - Should be fine if the underlying keypair remains the same

Reply:
We don't have to worry about the expiration of these certificates. cert-manager will auto-renew those.
Even if the `secret` is deleted it will just create a new secret with a fresh certificate.

Reply:
I don't know the max duration, but I was able to generate 100 years duration certificate.

Reply:
So my concern isn't on the Orc8r side -- I agree that everything should work fine, from a cert perspective, on Orc8r side. The issue actually is we manually copy the root cert to every AGW that will connect to the Orc8r.

This means that once the *original* root cert expires, the gateways will immediately view Orc8r controller certs as invalid. And even though Orc8r has renewed its root cert, we don't currently have a mechanism to propagate that renewed cert down to each gateway

Let me know if I'm missing something

Reply:
The possible resolution for this requires that only public CA signed certificates be used over self-signed (which cannot be automatically renewed). The classic flow would be to (a) Using an API, request for a public CA certificate that is used as the root to generate all the other keys. AWS Certificate Manager provides public CA certificates for free. (b) In the absence of an API based request or certificates already available, use public CA certificates from a local directory to generate all the other certificates (c) set expiry to 398 days for certificates (d) use cert manager to renew before expiry.